README.md

```diff
@@ -1,2 +1,178 @@
+   ```
+5. For more details on configuration and additional parameters for `reportportal.properties`,
+   refer [reportportal-client-java](https://github.com/reportportal/client-java)
+   and [reportportal-agent-java-testNG](https://github.com/reportportal/agent-java-testNG)
+
+# How to use the ReportPortal utilities
+
+1. `ReportPortalPropertiesOverloader` class takes care of dynamically setting Report Portal properties at run time like
+   launch name, description, attributes etc
+    1. Use `parameters.setLaunchName(<LaunchName>)` to set launch name
+    2. Use `parameters.setDescription(<LaunchDescription>)` to set launch description
+    3. Use `addAttributes(<Key>, <Value>)` to set any launch attribute. For instance,
+       `addAttributes("OS", System.getProperty("os.name"));`
+    4. **Dynamic properties:** For setting any additional property to launch attributes which is not already configured
+       in `ReportPortalPropertiesOverloader` class, set that attribute key and value at either System Property level or
+       at Environment variable. Any key with prefix `RP_` set at System property or Environment variable level will be
+       set as ReportPortal's launch attribute. The method which takes care of this configuration
+       is `setRPAttributesFromSystemVariables` defined inside `ReportPortalPropertiesOverloader` class.
+       For instance, if you're setting environment variable like `export RP_Version=0.0.1`, then on ReportPortal, you'll
+       see the corresponding launch attribute as `Version:0.0.1`
+2. `ReportPortalLogger` class has all its public methods defined as static, so it doesn't require the consumer to create
+   an object of this class. Following are the methods we can use to report logs to ReportPortal
+    1. `ReportPortalLogger.logInfoMessage(String message)`: Logs `message` at INFO level to `ReportPortal` as well as
+       console
+    2. `ReportPortalLogger.logDebugMessage(String message)`: Logs `message` at DEBUG level to `ReportPortal`
+    3. `ReportPortalLogger.logWarningMessage(String message)`: Logs `message` at WARN level to `ReportPortal`
+    4. `ReportPortalLogger.captureAndAttachScreenshot(WebDriver webDriver)`: This method implicitly
+       calls `captureScreenShot` method of `ScreenShotManager` class to capture the screenshot and then
+       calls `attachFileInReportPortal` method of `ReportPortalLogger` class to log the screenshot to ReportPortal
+    5. `ReportPortalLogger.captureAndAttachScreenshot(WebDriver webDriver, Sting message)`: It is overloading
+       the `captureScreenShot` method with additional capability of logging the message (passed in parameter) to
+       ReportPortal along with the screenshot
+    6. `ReportPortalLogger.attachFileInReportPortal(String message, File file)`: Logs `message` and attaches `file` at
+       INFO level to `ReportPortal`
+2. `ScreenShotManager` class has all its public methods defined as static, so it doesn't require the consumer to create
```

Comment:
This should not be required

Reply:
The class has one public method and rest details are merely for understanding on the implementation. I've changed the wordings to make it more clear that only method is public

Reply:
Removed documentation for all private methods